File: backend/src/main/java/com/onlinestore/backend/controller/InventoryController.java
package com.onlinestore.backend.controller;

import com.onlinestore.backend.model.Product;
import com.onlinestore.backend.repository.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/inventory")
public class InventoryController {

    @Autowired
    private ProductRepository productRepository;

    @GetMapping("/stats")
    public ResponseEntity<?> getInventoryStats() {
        List<Product> products = productRepository.findAll();

        int totalProducts = products.size();
        int totalStock = products.stream().mapToInt(p -> p.getStockQuantity() != null ? p.getStockQuantity() : 0).sum();
        double totalValue = products.stream()
                .mapToDouble(p -> (p.getPrice() != null ? p.getPrice() : 0.0) * (p.getStockQuantity() != null ? p.getStockQuantity() : 0))
                .sum();
        long lowStockCount = products.stream()
                .filter(p -> p.getStockQuantity() != null && p.getStockQuantity() > 0 && p.getStockQuantity() < 5)
                .count();
        long outOfStockCount = products.stream()
                .filter(p -> p.getStockQuantity() != null && p.getStockQuantity() == 0)
                .count();

        Map<String, Object> stats = new HashMap<>();
        stats.put("totalProducts", totalProducts);
        stats.put("totalStock", totalStock);
        stats.put("totalValue", totalValue);
        stats.put("lowStockCount", lowStockCount);
        stats.put("outOfStockCount", outOfStockCount);

        return ResponseEntity.ok(stats);
    }

    @GetMapping("/low-stock")
    public ResponseEntity<?> getLowStockProducts() {
        List<Product> lowStockProducts = productRepository.findByStockQuantityLessThanOrderByStockQuantityAsc(5);
        return ResponseEntity.ok(lowStockProducts);
    }
}

File: backend-old/controllers/inventoryController.js
const Product = require('../models/Product');

// @desc    Get overall inventory statistics
// @route   GET /api/inventory/stats
// @access  Private/Admin
const getInventoryStats = async (req, res) => {
    try {
        const products = await Product.find({});
        
        const totalProducts = products.length;
        const totalStock = products.reduce((acc, p) => acc + p.stockQuantity, 0);
        const totalValue = products.reduce((acc, p) => acc + (p.price * p.stockQuantity), 0);
        const lowStockCount = products.filter(p => p.stockQuantity > 0 && p.stockQuantity < 5).length;
        const outOfStockCount = products.filter(p => p.stockQuantity === 0).length;

        res.json({
            totalProducts,
            totalStock,
            totalValue,
            lowStockCount,
            outOfStockCount
        });
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};

// @desc    Get products with low stock (< 5)
// @route   GET /api/inventory/low-stock
// @access  Private/Admin
const getLowStockProducts = async (req, res) => {
    try {
        // Find products where stock is less than 5
        const lowStockProducts = await Product.find({ stockQuantity: { $lt: 5 } }).sort({ stockQuantity: 1 });
        res.json(lowStockProducts);
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};

module.exports = {
    getInventoryStats,
    getLowStockProducts
};
